src/routes.js

```diff
@@ -29,10 +26,7 @@ exports.SEARCH_PAGE = async (page, request, query, requestQueue, maxPostCount, e
             '#debug': Apify.utils.createRequestDebugInfo(request),
         });
     }
-    // sometimes it loads totally different page. In this case number of items would be > 22. Need to retry.
-    if (resultsLength > 22) {
-        throw new Error("Page didn't load properly, retrying...");
-    }
```

Comment:
Are you sure this is ok to remove?

Reply:
Yes, it throws an error and the exact error is reported in #33 and #34 .